cmake/modules/AddSwift.cmake

```diff
@@ -434,18 +438,35 @@ function(_add_variant_link_flags)
     list(APPEND link_libraries "bsd" "atomic")
     list(APPEND result "-Wl,-Bsymbolic")
   elseif("${LFLAGS_SDK}" STREQUAL "ANDROID")
-    list(APPEND link_libraries "dl" "log" "atomic" "icudataswift" "icui18nswift" "icuucswift")
+    if("${SWIFT_HOST_VARIANT}" STREQUAL "android")
+      list(APPEND link_libraries "dl" "log" "atomic" "icudata" "icui18n" "icuuc")
+    else()
+      list(APPEND link_libraries "dl" "log" "atomic" "icudataswift" "icui18nswift" "icuucswift")
+    endif()
     # We provide our own C++ below, so we ask the linker not to do it. However,
     # we need to add the math library, which is linked implicitly by libc++.
     list(APPEND result "-nostdlib++" "-lm")
-    if("${LFLAGS_ARCH}" MATCHES armv7)
-      set(android_libcxx_path "${SWIFT_ANDROID_NDK_PATH}/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a")
-    elseif("${LFLAGS_ARCH}" MATCHES aarch64)
-      set(android_libcxx_path "${SWIFT_ANDROID_NDK_PATH}/sources/cxx-stl/llvm-libc++/libs/arm64-v8a")
+    if("${SWIFT_HOST_VARIANT}" STREQUAL "android")
+      set(android_libcxx_path "${ANDROID_NATIVE_SYSROOT}/usr/lib")
+      if("${LFLAGS_ARCH}" MATCHES armv7)
+        list(APPEND library_search_directories "/system/lib")
+      elseif("${LFLAGS_ARCH}" MATCHES aarch64)
+        list(APPEND library_search_directories "/system/lib64")
+      else()
+        message(SEND_ERROR "unknown architecture (${LFLAGS_ARCH}) when natively compiling on android")
+      endif()
     else()
-      message(SEND_ERROR "unknown architecture (${LFLAGS_ARCH}) for android")
+     if("${LFLAGS_ARCH}" MATCHES armv7)
+       set(android_libcxx_path "${SWIFT_ANDROID_NDK_PATH}/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a")
+     elseif("${LFLAGS_ARCH}" MATCHES aarch64)
+       set(android_libcxx_path "${SWIFT_ANDROID_NDK_PATH}/sources/cxx-stl/llvm-libc++/libs/arm64-v8a")
+     else()
+       message(SEND_ERROR "unknown architecture (${LFLAGS_ARCH}) when cross-compiling for android")
+     endif()
+    endif()
+    if(NOT "${SWIFT_HOST_VARIANT}" STREQUAL "android")
+      list(APPEND link_libraries "${android_libcxx_path}/libc++abi.a")
```

Comment:
I think that the existing paths should work or the android native compilation as well.

Reply:
Not sure what you mean, the NDK uses a different path layout than the Termux app.

Reply:
We should figure out how to map the paths appropriately and use that to have a specialised build.  Alternatively, we can refactor this out into a toolchain.

Reply:
>We should figure out how to map the paths appropriately and use that to have a specialised build.

Isn't that what I'm doing here? I'm specifying different paths based on whether it's the NDK being used or the Termux app.

>Alternatively, we can refactor this out into a toolchain.

It's easier to just build off this existing work, but that may make sense at some point.